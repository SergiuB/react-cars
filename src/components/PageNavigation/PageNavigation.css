.page-navigation button {
  margin: 5px;
}