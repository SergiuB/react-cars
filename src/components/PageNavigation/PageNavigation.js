const PageNavigation = () => null;

export default PageNavigation;
