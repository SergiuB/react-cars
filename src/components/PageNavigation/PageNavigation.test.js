import React from 'react';
import { shallow } from 'enzyme';

import PageNavigation from './PageNavigation';

describe('PageNavigation', function () {
  it('renders correctly with default props', function () {
    const wrapper = shallow(
      <PageNavigation />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with buttons disabled', function () {
    const wrapper = shallow(
      <PageNavigation nextEnabled={false} prevEnabled={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('fires onNext when next btn is clicked', function () {
    const onNext = jest.fn();
    const wrapper = shallow(
      <PageNavigation onNext={onNext} />
    );
    wrapper.find('[name="next"]').simulate('click');
    expect(onNext).toHaveBeenCalled();
  });

  it('fires onPrev when prev btn is clicked', function () {
    const onPrev = jest.fn();
    const wrapper = shallow(
      <PageNavigation onPrev={onPrev} />
    );
    wrapper.find('[name="prev"]').simulate('click');
    expect(onPrev).toHaveBeenCalled();
  });
});
