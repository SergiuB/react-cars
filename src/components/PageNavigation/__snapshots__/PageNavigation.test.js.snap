// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigation renders correctly with buttons disabled 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="page-navigation"
>
    <button
        disabled={true}
        name="prev"
        onClick={[Function]}
    >
        Prev
    </button>
    <button
        disabled={true}
        name="next"
        onClick={[Function]}
    >
        Next
    </button>
</div>,
  "nodes": Array [
    <div
      className="page-navigation"
>
      <button
            disabled={true}
            name="prev"
            onClick={[Function]}
      >
            Prev
      </button>
      <button
            disabled={true}
            name="next"
            onClick={[Function]}
      >
            Next
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PageNavigation
        nextEnabled={false}
        onNext={[Function]}
        onPrev={[Function]}
        prevEnabled={false}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "nextEnabled": false,
          "onNext": [Function],
          "onPrev": [Function],
          "prevEnabled": false,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="page-navigation"
>
          <button
                    disabled={true}
                    name="prev"
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    disabled={true}
                    name="next"
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="page-navigation"
>
          <button
                    disabled={true}
                    name="prev"
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    disabled={true}
                    name="next"
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PageNavigation
    nextEnabled={false}
    onNext={[Function]}
    onPrev={[Function]}
    prevEnabled={false}
/>,
}
`;

exports[`PageNavigation renders correctly with default props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="page-navigation"
>
    <button
        disabled={false}
        name="prev"
        onClick={[Function]}
    >
        Prev
    </button>
    <button
        disabled={false}
        name="next"
        onClick={[Function]}
    >
        Next
    </button>
</div>,
  "nodes": Array [
    <div
      className="page-navigation"
>
      <button
            disabled={false}
            name="prev"
            onClick={[Function]}
      >
            Prev
      </button>
      <button
            disabled={false}
            name="next"
            onClick={[Function]}
      >
            Next
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PageNavigation
        nextEnabled={true}
        onNext={[Function]}
        onPrev={[Function]}
        prevEnabled={true}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "nextEnabled": true,
          "onNext": [Function],
          "onPrev": [Function],
          "prevEnabled": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="page-navigation"
>
          <button
                    disabled={false}
                    name="prev"
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    disabled={false}
                    name="next"
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="page-navigation"
>
          <button
                    disabled={false}
                    name="prev"
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    disabled={false}
                    name="next"
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PageNavigation
    nextEnabled={true}
    onNext={[Function]}
    onPrev={[Function]}
    prevEnabled={true}
/>,
}
`;
